lib/utils/dom.js

```diff
@@ -38,6 +38,18 @@ export function observe(ele: Node, options: MutationObserverInit, callback: Iter
 	return observer;
 }
 
+export function observeNewThing(ele: Node): Promise<MutationRecord> {
+	return new Promise(resolve => {
+		const observer = observe(ele, { childList: true }, mutation => {
+			if (Array.from(mutation.addedNodes).find(e => e.classList.contains('thing'))) {
+				observer.disconnect();
+				resolve(mutation);
```

Comment:
I'm cool with that. Probably makes the most sense, as the calling code should most likely only need access to the target (Thing container)